Dockerfile

```diff
@@ -292,3 +292,10 @@ RUN bash /opt/terraform/helpers/build
 ENV HOME="/home/dependabot"
 
 WORKDIR ${HOME}
+
+# Place a git shim ahead of git on the path to rewrite git arguments to use HTTPS.
+ARG SHIM="https://github.com/dependabot/git-shim/releases/download/v1.4.0/git-v1.4.0-linux-amd64.tar.gz"
+RUN curl -sL $SHIM -o git-shim.tar.gz && mkdir -p ~/bin && tar -xvf git-shim.tar.gz -C ~/bin && rm git-shim.tar.gz
+ENV PATH="$HOME/bin:$PATH"
+# Configure cargo to use git CLI so the above takes effect
+RUN mkdir -p ~/.cargo && printf "[net]\ngit-fetch-with-cli = true\n" >> ~/.cargo/config.toml
```

Comment:
You mentioned in the PR description that although this is a bit of a hack, it'll allow us to replace rewriting in multiple places in our code, which is a definite 🏆 . But to do that, all native helpers will need to be reconfigured to query git via cli and not native wire libraries... from those that you saw, did that option look possible in those helpers? Otherwise it'll prevent some of that 🏅...

Reply:
Totally out of scope of this PR, but maybe we should file an issue to eventually reach out to all the native helpers that we need to shim to ask them to add an option for natively use https urls... then we wouldn't need the shim here, but would instead simply pass `[net]\ngit-fetch-with-https = true` here.

If you don't have time / interest, I can reach out to those ecosystems to get the conversations started, but if you could at least file the dependabot-side issue listing all the ecosystems that we are currently patching that'd give me a checklist to start with and something I could cross-link.

Reply:
As a first pass I was going to get this cargo issue fixed, and then we can start removing hacks in a controlled manner per-ecosystem. I did search a good amount for an option in cargo to force https but there isn't one currently. 

That's a great idea to reach out to ask for that option in the package managers. My other thought was we could see about improving `insteadOf` functionality in Git. We have some Get contributors here at GitHub who might be able to help. However I struggled with how that would work exactly, so I went this shim route.

Reply:
> My other thought was we could see about improving insteadOf functionality in Git. We have some Get contributors here at GitHub who might be able to help. 

Yeah, that'd be an even better idea.

> However I struggled with how that would work exactly, so I went this shim route.

To clarify, are you unclear how to discover/work with the git contributors at GitHub to fix this upstream? Or even if `insteadOf` were improved to cover all these cases, do you still think we'd need the shim?